firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -3401,6 +3406,101 @@ public void onComplete(DatabaseError error, DatabaseReference ref) {
     IntegrationTestHelpers.waitFor(semaphore);
   }
 
+  @Test
+  public void emptyQueryGet() throws DatabaseException, InterruptedException {
+    DatabaseReference node = IntegrationTestHelpers.getRandomNode();
+    DatabaseConfig cfg = IntegrationTestHelpers.newTestConfig();
+    final Semaphore semaphore = new Semaphore(0);
+    node.get()
+        .addOnCompleteListener(
+            new OnCompleteListener<DataSnapshot>() {
+              @Override
+              public void onComplete(@NonNull Task<DataSnapshot> task) {
+                assertTrue(task.isSuccessful());
+                assertNotNull(task.getException());
+                assertEquals(task.getException().getMessage(), "Client offline with empty cache!");
+                semaphore.release();
+              }
+            });
+    IntegrationTestHelpers.waitFor(semaphore);
+  }
+
+  @Test
+  public void offlineQueryGet() throws DatabaseException {
+    DatabaseConfig cfg = IntegrationTestHelpers.newTestConfig();
+    IntegrationTestHelpers.goOffline(cfg);
+    DatabaseReference node = IntegrationTestHelpers.getRandomNode();
+    node.get()
+        .addOnCompleteListener(
+            new OnCompleteListener<DataSnapshot>() {
+              @Override
+              public void onComplete(@NonNull Task<DataSnapshot> task) {
+                Log.d("QueryTest", "offlineQueryGet onCompleteListener running.");
+                assertFalse(task.isSuccessful());
+                assertNotNull(task.getException());
+                assertEquals(task.getException().getMessage(), "Client offline with empty cache!");
+              }
+            });
+  }
+
+  @Test
+  public void getQueryBasic() throws DatabaseException, InterruptedException {
+    DatabaseReference ref = IntegrationTestHelpers.getRandomNode();
+    final Semaphore semaphore = new Semaphore(0);
+    ref.setValue(42)
```

Comment:
If calling `.get()` works for these tests, this could be as simple as

`ref.setValue(42).get()` followed by `ref.get().get()`.